scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/services/GraphQLMerlinService.java

```diff
@@ -766,6 +775,274 @@ public DatasetId storeSimulationResults(final PlanMetadata planMetadata,
+    final var unfinishedActivities = new HashMap<SimulatedActivityId, UnfinishedActivity>();
+    for(final var unfinishedActivityJson: unfinishedActivitiesJson){
+      final var activityAttributes = activityAttributesP.parse(unfinishedActivityJson.asJsonObject().getJsonObject("attributes")).getSuccessOrThrow();
+      SimulatedActivityId parentId = null;
+      if(!unfinishedActivityJson.asJsonObject().isNull("parent_id")){
+        parentId = new SimulatedActivityId(unfinishedActivityJson.asJsonObject().getJsonNumber("parent_id").longValue());
+      }
+      final var activityType = unfinishedActivityJson.asJsonObject().getJsonString("type").getString();
+      final var start = instantFromStart(simulationStart,
+          durationFromPGInterval(unfinishedActivityJson.asJsonObject().getJsonString("start_offset").getString()));
+      final var id = new SimulatedActivityId(unfinishedActivityJson.asJsonObject().getJsonNumber("id").longValue());
+      Optional<ActivityDirectiveId> actDirectiveId = Optional.empty();
+      if(activityAttributes.directiveId().isPresent()){
+        actDirectiveId = Optional.of(new ActivityDirectiveId(activityAttributes.directiveId().get()));
+      }
+      final var unfinishedActivity = new UnfinishedActivity(
+          activityType,
+          activityAttributes.arguments(),
+          start,
+          parentId,
+          List.of(),
+          actDirectiveId
+      );
+      unfinishedActivities.put(id, unfinishedActivity);
+    }
+    return unfinishedActivities;
+  }
+
+  private record Profiles(
+      Map<String, Pair<ValueSchema, List<ProfileSegment<RealDynamics>>>> realProfiles,
+      Map<String, Pair<ValueSchema, List<ProfileSegment<SerializedValue>>>> discreteProfiles
+  ){}
+
+  private Profiles parseProfiles(JsonArray dataset){
+    Map<String, Pair<ValueSchema, List<ProfileSegment<RealDynamics>>>> realProfiles = new HashMap<>();
+    Map<String, Pair<ValueSchema, List<ProfileSegment<SerializedValue>>>> discreteProfiles = new HashMap<>();
+    for(final var profile:dataset){
+      final var duration = durationFromPGInterval(profile.asJsonObject().getString("duration"));
+      final var name = profile.asJsonObject().getString("name");
+      final var type = profile.asJsonObject().getJsonObject("type");
+      final var typetype = type.getString("type");
+      final boolean isReal = typetype.equals("real");
+      if(isReal){
+        final var realProfile = parseProfile(profile.asJsonObject(), duration, realDynamicsP);
+        realProfiles.put(name, realProfile);
+      } else {
+        final var discreteProfile = parseProfile(profile.asJsonObject(), duration, serializedValueP);
+        discreteProfiles.put(name, discreteProfile);
+      }
+    }
+    return new Profiles(realProfiles, discreteProfiles);
+  }
+
+  public <Dynamics> Pair<ValueSchema, List<ProfileSegment<Dynamics>>> parseProfile(JsonObject profile, Duration profileExtent, JsonParser<Dynamics> dynamicsParser){
```

Comment:
How does what this function does differ from `realProfileP.parse(profile).getSuccessOrThrow()`/`discreteProfileP.parse(profile).getSuccessOrThrow()`?

Reply:
It correctly offsets the times of the segments with regard to the total duration of the profile.

Reply:
Veri similarly to what's done in `GetProfileSegmentsAction`

Reply:
Understood. Can you add a brief comment clarifying that this method is necessary in order to get the duration, since the `duration` column in the DB is not equal to the local "extent" (what we need here)?